### PR TITLE
fix(nexus): fixing missing I/Os during nexus rebuild

### PR DESCRIPTION
**Pull request by dsavitskiy**
If a nexus was published after a rebuild started, the child being rebuilt
was not added to writers. This caused I/Os missing the child, leading to
a corrupted replica.

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

File: scripts/rust-linter.sh [hunk 1]
```diff
@@ -9,4 +9,5 @@ $CARGO clippy --all --all-targets -- -D warnings \
     -A clippy::option_env_unwrap \
     -A clippy::redundant-guards \
     -A clippy::suspicious-doc-comments \
+    -A clippy::useless-format \
```
Comment (tiagolobocastro) [inline scripts/rust-linter.sh:12]: why do we need this?
Btw seems like we have a lot of whitelists here, any reason for this?

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: scripts/rust-linter.sh (hunk 1 above)
Comment (dsavitskiy) [reply scripts/rust-linter.sh:12]: As we update rustc/clippy versions, new lints come in. Some of them are not easily fixable, and for some we just don't have enough time.

As for "useless-format", it is a newer one and a quite annoying.

E.g. you have a list like:
let args = [
    format!("--abc={a}"),
    format!("--def"),
    format!("--qwe={b}"),
];

It is nicely and uniformly formatted, and does not cause any troubles. Now, clippy started to complain about it, suggesting to replace format!("--def") with "--def".to_string(). I don't see much sense in this.

**Review by hrudaya21 — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**